oauthenticator/oauth2.py

```diff
@@ -264,6 +276,49 @@ class OAuthenticator(Authenticator):
         """,
     )
 
+    claim_groups_key = Union(
```

Comment:
I don't really want to start a naming conversation when this is just moving existing config up one level, but I think this config name is very unclear, and defining new config on a class is an opportunity to _not_ inherit the confusing name. `claim` and `key` are really synonyms, so it's odd to have both at all, let alone one on either side of the field they populate. Could it be just `groups_key` (generic, Pythonic) or `groups_claim` (OIDC-specific, jargony, I suspect unclear to most) or `userdata_groups_key` to more clearly specify that it's the key _in userdata_ where this info will be found?

If we rename, Generic can keep its `claims_group_key` as a deprecated trait that assigns to `self.userdata_groups_key` with a warning.

Reply:
`userdata_groups_key` sounds good to me as well

Reply:
The base one is called `auth_state_groups_key`, as the direct analog to `userdata_groups_key`, since now it gets info out of `auth_state` instead of `userdata`. Generic keeps its `claims_group_key` and issues a warning.